src/Nancy.Hosting.Aspnet/NancyHandler.cs

```diff
@@ -114,6 +114,18 @@ private static Request CreateNancyRequest(HttpContextBase context)
                 protocolVersion);
         }
 
+        private static bool IsChunkEncoded(IDictionary<string, IEnumerable<string>> incomingHeaders)
+        {
+            if (incomingHeaders == null)
+            {
+                return false;
+            }
+            IEnumerable<string> transferEncodingValue;
+            var isTransferEncoded = incomingHeaders.TryGetValue("Transfer-Encoding", out transferEncodingValue);
+            var transferEncodingString = isTransferEncoded ? transferEncodingValue.SingleOrDefault() : string.Empty;
+            return transferEncodingString != null && transferEncodingString.ToLower().Equals("chunked");

```

Comment:
Let's compare this using proper casing, instead of doing `ToLower` and allocating an extra string.

```csharp
string.Equals(transferEncodingString , "chunked", StringComparison. OrdinalIgnoreCase)
```